### DISCUSSION TITLE
Add an option to disable typescript validation for VS Code extension

### DISCUSSION THREAD

**Discussion by vrabe** [withastro/roadmap#984] [Proposal]
### Body

# Summary

Add an option to disable TypeScript validation for VS Code extension.

# Background & Motivation

Currently, Astro VS Code extension reports TypeScript problems in `*.astro` files. But some TypeScript validation rules are not applicable for JavaScript code. It's good for JavaScript coders to have an option to disable TypeScript validation so that they don't have to ignore false-positive problems show in VS Code.

Although there is a workaround: `// @ts-nocheck`, it's unrealistic and bad-looking to have `// @ts-nocheck` in all `*.astro` files contain JavaScript code.

There was a discussion about this: withastro/language-tools#604. 

# Goals

- Have an option on Astro VS Code extension which users can disable TypeScript validation.

# Example

N/A

**Comment by Princesseuh** [comment]
You can use `typescript.validate` to disable TypeScript validation in the editor, fwiw.

**Reply by vrabe** [reply]
It works! Thanks.
